src/transformers/models/clvp/modeling_clvp.py

```diff
@@ -0,0 +1,1873 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch CLVP model."""
+
+
+import math
+from dataclasses import dataclass
+from typing import Optional, Tuple, Union
+
+import torch
+import torch.utils.checkpoint
+from torch import nn
+from torch.nn import CrossEntropyLoss
+
+from transformers.generation import GenerationConfig
+
+from ...activations import ACT2FN
+from ...modeling_outputs import BaseModelOutput, BaseModelOutputWithPooling, CausalLMOutputWithCrossAttentions
+from ...modeling_utils import PreTrainedModel, SequenceSummary
+from ...pytorch_utils import Conv1D
+from ...utils import (
+    ModelOutput,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    logging,
+    replace_return_docstrings,
+)
+from .configuration_clvp import (
+    CLVPAutoRegressiveConfig,
+    CLVPConfig,
+    CLVPSpeechConfig,
+    CLVPTextConfig,
+    PretrainedConfig,
+)
+
+
+logger = logging.get_logger(__name__)
+
+_CHECKPOINT_FOR_DOC = "susnato/clvp_dev"
+
+CLVP_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "susnato/clvp_dev",
```

Comment:
Do you need us to help you transfer this to the original organisation

Reply:
Yes, but let's do this at the end of the `Tortoise-TTS` integration, since we need to transfer all files of `clvp`, `univnet` and the `diffusion_decoder` at some point. Better to do it once.